pyvista/plotting/helpers.py

```diff
@@ -99,11 +109,11 @@ def plot(var_item, off_screen=None, full_screen=False, screenshot=None,
 
     eye_dome_lighting = kwargs.pop("edl", eye_dome_lighting)
     show_grid = kwargs.pop('show_grid', False)
```

Comment:
Ok, that might be why you did not want the IGNORE-option for Themes or see there usecase. I guess allowing settings via kwargs is how it is done up till now. I would have liked to remove these kwargs parameter setting in favor of themes. If I understand it correctly, now we have two ways to change settings, by giving a theme-parameter or by changing individual settings? If the latter is how it was done up till now, I guess removing it and "enforce" the use of themes would be quit a big compatibility breaker, so thats maybe not an option ^^

Reply:
Not sure I understand your point. We could have kwargs -> local theme -> global theme with no issues. If you pass a theme and a kwarg then the kwarg comes first. That's fairly intuitive I think.

It would be as simple as `kwargs.pop('show_grid', self.get_default('show_grid'))`. where `self.get_default` looks up `self.theme` and then `pyvista.theme` if necessary.

Reply:
> It would be as simple as kwargs.pop('show_grid', self.get_default('show_grid'))

I guess thats ok - I simply don't like the hidden `kwargs` that are not known to the user (or at least he has to dig deep inside the code or documentation). Using a class-object is imho clearer (but I admitt slightly more code for simple changes). Thats also true on the implementation side, because its not easy to make shure that all `kwargs` are handled correctly/completly due to using them in several different places and passing them through several methods. There might even occur an unwanted conflict, because of using a method parameter thats already optional inside `kwargs` (as kwargs are optional, such a conflict might become hard to spot and seemingly random). Maybe it would be possible to handle the `kwargs` at a single place e.g. as a function or even a method of `Theme`, thats called, whenever the used theme (default or custom) is finalized, to change that finalized theme. That would only work, if `BasePlotter` (and maybe also some other places) create a local copy of the theme. But for the implementation it would have the advantage, that whenever a setting is needed only the `self._theme` would be used.

Reply:
I keep feeling there are multiple different design discussions being intertwined here. In this case I can put my finger on
  1. whether to allow the IGNORE theme values,
  2. whether individual keyword parameters can _also_ be used in addition to a theme,
  3. whether `**kwargs` is kept in the signature.
  
The three are mostly independent. Number 1 seems reasonable to me. Number 2 should absolutely be the case, I don't want to instantiate a `Theme` so that I can pass a single keyword argument I need. Number 3 can be up for debate, but I suspect most use cases involving them have to do with
  1. aliases (like `'edl'` vs `'eye_dome_lighting'` above). Note that this case doesn't hinder discoverability, because there is a corresponding explicit name in the signature (`eye_dome_lighting=False` above).
  2. passing through kwargs for lower-level objects (not sure if we actually have any of these in pyvista)

In the above snippet `show_grid` could be made into a full-fledged keyword argument if necessary.

Reply:
2. fully agree (I have no issue if it is a true parameter instead of hidden kwargs)
3. might be off topic for this PR

> passing through kwargs for lower-level objects (not sure if we actually have any of these in pyvista)

Just as an example Plotter.show() has a `kwarg`. There is an `assert_empty_kwargs(**kwargs)` to make sure no wrong parameter are passed. But before that `jupyter_kwargs` is extracted from `kwargs`, which is later passsed to `handle_plotter()` where it is passed on to `show_ipygany()` - for me thats quite confusing. But you are right, pyvista uses `kwargs` not as much as what I thought.